File: protocol/CSharp/ProtoCommon.cs
 
 class STATIC_BEC_SUCCESS //成功 
 { 
 	public static Proto4z.ui16 value = new Proto4z.ui16(0);  
 } 
 class STATIC_BEC_UNKOWN_ERROR //未知错误 
 { 
 	public static Proto4z.ui16 value = new Proto4z.ui16(1);  
 } 
 class STATIC_BEC_INNER_ERROR //内部错误 
 { 
 	public static Proto4z.ui16 value = new Proto4z.ui16(2);  
 } 
 class STATIC_BEC_DB_ERROR //数据库错误 
 { 
 	public static Proto4z.ui16 value = new Proto4z.ui16(3);  
 } 
 class STATIC_BEC_PARAM_DENIED //非法参数 
 { 
 	public static Proto4z.ui16 value = new Proto4z.ui16(4);  
 } 
 class STATIC_BEC_PERMISSION_DENIED //权限非法 
 { 
 	public static Proto4z.ui16 value = new Proto4z.ui16(5);  
 } 
 class STATIC_BEC_SYSTEM_ERROR //系统错误 
 { 
 	public static Proto4z.ui16 value = new Proto4z.ui16(6);  
 } 
 class STATIC_BEC_INVALIDE_USERID //用户ID无效 
 { 
 	public static Proto4z.ui16 value = new Proto4z.ui16(50);  
 } 
 class STATIC_BEC_AUTH_USER_NOT_EXIST //认证错误:用户不存在 
 { 
 	public static Proto4z.ui16 value = new Proto4z.ui16(60);  
 } 
 class STATIC_BEC_AUTH_PASSWD_INCORRECT //认证错误:密码错误 
 { 
 	public static Proto4z.ui16 value = new Proto4z.ui16(61);  
 } 
 class STATIC_BEC_AUTH_ING //认证错误:正在认证中... 
 { 
 	public static Proto4z.ui16 value = new Proto4z.ui16(62);  
 } 
 class STATIC_BEC_AUTH_AREADY_AUTH //认证错误:已认证 
 { 
 	public static Proto4z.ui16 value = new Proto4z.ui16(63);  
 } 
 class STATIC_BEC_AUTH_LIMITE_COUNT //认证错误:认证次数超过限制 
 { 
 	public static Proto4z.ui16 value = new Proto4z.ui16(64);  
 } 
  
 class UserInfo: Proto4z.IProtoObject //用户信息 
 {	 
 	public Proto4z.ui64 uid;  
 	public Proto4z.String nickName; //用户昵称 
 	public Proto4z.i32 iconID; //头像 
 	public Proto4z.i32 level; //等级 
 	public Proto4z.i32 diamond; //当前剩余的充值钻石 
 	public Proto4z.i32 hisotryDiamond; //历史充值钻石总额 
 	public Proto4z.i32 giftDiamond; //当前剩余的赠送钻石 
 	public Proto4z.ui32 joinTime; //加入时间 
 	public System.Collections.Generic.List<byte> __encode() 
 	{ 
 		var ret = new System.Collections.Generic.List<byte>(); 
 		ret.AddRange(uid.__encode()); 
 		ret.AddRange(nickName.__encode()); 
 		ret.AddRange(iconID.__encode()); 
 		ret.AddRange(level.__encode()); 
 		ret.AddRange(diamond.__encode()); 
 		ret.AddRange(hisotryDiamond.__encode()); 
 		ret.AddRange(giftDiamond.__encode()); 
 		ret.AddRange(joinTime.__encode()); 
 		return ret; 
 	} 
 	public System.Int32 __decode(byte[] binData, ref System.Int32 pos) 
 	{ 
 		uid = new ui64(); 
 		uid.__decode(binData, ref pos); 
 		nickName = new string(); 
 		nickName.__decode(binData, ref pos); 
 		iconID = new i32(); 
 		iconID.__decode(binData, ref pos); 
 		level = new i32(); 
 		level.__decode(binData, ref pos); 
 		diamond = new i32(); 
 		diamond.__decode(binData, ref pos); 
 		hisotryDiamond = new i32(); 
 		hisotryDiamond.__decode(binData, ref pos); 
 		giftDiamond = new i32(); 
 		giftDiamond.__decode(binData, ref pos); 
 		joinTime = new ui32(); 
 		joinTime.__decode(binData, ref pos); 
 	} 
 } 
  
 class UserInfoVct : System.Collections.Generic.List<Proto4z.UserInfo>, IProtoObject  
 { 
 	public System.Collections.Generic.List<byte> __encode() 
 	{ 
 		var ret = new System.Collections.Generic.List<Byte>(); 
 		 var len = new Proto4z.ui16((UInt16)this.Count); 
 		ret.AddRange(len.__encode()); 
 		for (int i = 0; i < this.Count; i++ ) 
 		{ 
 			ret.AddRange(this[i].__encode()); 
 		} 
 		return ret; 
 	} 
  
 	public System.Int32 __decode(byte[] binData, ref System.Int32 pos) 
 	{ 
 		var len = new Proto4z.ui16(0); 
 		len.__decode(binData, ref pos); 
 		if(len.val > 0) 
 		{ 
 			for (int i=0; i<len.val; i++) 
 			{ 
 				var data = new Proto4z.UserInfo(); 
 				 data.__decode(binData, ref pos); 
 				this.Add(data); 
 			} 
 		} 
 		return pos; 
 	} 
 } 
  
  
 

File: protocol/lua/ProtoCommon.lua
Protoz.BEC_SUCCESS = 0--成功 
 Protoz.BEC_UNKOWN_ERROR = 1--未知错误 
 Protoz.BEC_INNER_ERROR = 2--内部错误 
 Protoz.BEC_DB_ERROR = 3--数据库错误 
 Protoz.BEC_PARAM_DENIED = 4--非法参数 
 Protoz.BEC_PERMISSION_DENIED = 5--权限非法 
 Protoz.BEC_SYSTEM_ERROR = 6--系统错误 
 Protoz.BEC_INVALIDE_USERID = 50--用户ID无效 
 Protoz.BEC_AUTH_USER_NOT_EXIST = 60--认证错误:用户不存在 
 Protoz.BEC_AUTH_PASSWD_INCORRECT = 61--认证错误:密码错误 
 Protoz.BEC_AUTH_ING = 62--认证错误:正在认证中... 
 Protoz.BEC_AUTH_AREADY_AUTH = 63--认证错误:已认证 
 Protoz.BEC_AUTH_LIMITE_COUNT = 64--认证错误:认证次数超过限制 
  
 Protoz.UserInfo = {} --用户信息 
 Protoz.UserInfo.__getName = "UserInfo" 
 Protoz.UserInfo[1] = {name="uid", type="ui64"}  
 Protoz.UserInfo[2] = {name="nickName", type="string"} --用户昵称 
 Protoz.UserInfo[3] = {name="iconID", type="i32"} --头像 
 Protoz.UserInfo[4] = {name="level", type="i32"} --等级 
 Protoz.UserInfo[5] = {name="diamond", type="i32"} --当前剩余的充值钻石 
 Protoz.UserInfo[6] = {name="hisotryDiamond", type="i32"} --历史充值钻石总额 
 Protoz.UserInfo[7] = {name="giftDiamond", type="i32"} --当前剩余的赠送钻石 
 Protoz.UserInfo[8] = {name="joinTime", type="ui32"} --加入时间 
  
 Protoz.UserInfoVct = {}  
 Protoz.UserInfoVct.__getName = "UserInfoVct" 
 Protoz.UserInfoVct.__getDesc = "array" 
 Protoz.UserInfoVct.__getTypeV = "UserInfo" 
 